returnn/tf/updater.py

```diff
@@ -47,25 +49,37 @@ def _init_optimizer_classes_dict():
       continue
     if v is Optimizer:
       continue
+    if tf_compat.v2 and v is KerasOptimizer:
+      continue
     if not isinstance(v, type):
       continue
     if not issubclass(v, allowed_types):
       continue
+    if v is KerasOptimizerWrapper:
+      continue
     register_optimizer_class(v, name=name)
 
 
+def _check_valid_optimizer(optimizer_class):
+  """
+  :param type optimizer_class:
+  """
+  if tf_compat.v2:
+    assert (issubclass(optimizer_class, tf_compat.v2.keras.optimizers.Optimizer) or
+            issubclass(optimizer_class, Optimizer))
```

Comment:
Do not use hanging indents.
Also, use `KerasOptimizer` directly in the `issubclass` check.
Also, is it not possible to directly specify the tuple in `issubclass`, like you would do for `isinstance`?
Also, check for `KerasOptimizer` instead of `tf_compat.v2`.
So:
```suggestion
  if KerasOptimizer:
    assert issubclass(optimizer_class, (KerasOptimizer, Optimizer))
```